components/dashboards-web-component/src/designer/DashboardDesigner.jsx

```diff
@@ -265,11 +304,82 @@ export default class DashboardDesigner extends Component {
                 pageId = this.state.dashboard.landingPage;
             }
 
-            let dashboard = this.state.dashboard;
+            let dashboard = this.getDashboard();
+
+            let optionsMap = {};
+
+            /**
+             * helper function to get the page from a given dashboard given the pageId
+             * */
+            function getPage(dashboard, pageID) {
+                let pages = dashboard.pages;
+                if (pages) {
+                    for (let i = 0; i < pages.length; i++) {
+                        let page = pages[i];
+                        if (page.id === pageID) {
+                            return page;
+                        }
+                    }
+                }
+                return null;
+            }
+
+            let page = getPage(dashboard, pageId);
+
+            /**
+             * helper function to traverse the content of a page and find widgets to extract their options to a
+             * given map
+             * @param page : the page to be traversed
+             * @param OptionsMap : the map to extract option configurations
+             * */
+            function searchForWidgetsInAPage(page, OptionsMap) {
+                let obj = page;
+                if (obj.type && obj.type === "component" && obj.props && obj.props.id) {
+                    if (obj.props.configs.options) {
+                        let newObj = JSON.parse(JSON.stringify(obj));
+                        OptionsMap[obj.props.id] = newObj.props.configs.options;
+                    }
+                }
+                else if (obj.content) {
```

Comment:
Move the `else if` to the previous line.
`} else if (obj.content) {`

Reply:
I think it's better to keep it this way for readability and clarity

Reply:
The suggested way was used across carbon-dashboards. Please refer [[1]](https://github.com/NisalaNiroshana/carbon-dashboards/blob/1a44245d678dfd0523afdeaa1e9ca242bdf1e598/components/dashboards-web-component/src/viewer/DashboardView.jsx#L202), [[2]](https://github.com/NisalaNiroshana/carbon-dashboards/blob/1a44245d678dfd0523afdeaa1e9ca242bdf1e598/components/dashboards-web-component/src/viewer/components/Sidebar.jsx#L87) and [[3]](https://github.com/NisalaNiroshana/carbon-dashboards/blob/025420a0e0b57df9003150057559a19cdeda9719/components/dashboards-web-component/src/designer/components/PagesPanel.jsx#L201) for more clarifications.